core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -165,12 +124,154 @@ object ConsumerGroupCommand extends Logging {
                                                 consumerId: Option[String], host: Option[String],
                                                 clientId: Option[String], logEndOffset: Option[Long])
 
+  protected case class MemberAssignmentState(group: String, consumerId: String, host: String, clientId: String,
+                                             numPartitions: Int, assignment: Option[List[TopicPartition]])
+
+  protected case class GroupState(group: String, assignmentStrategy: String, state: String, numMembers: Int)
+
   sealed trait ConsumerGroupService {
 
     def listGroups(): List[String]
 
-    def describeGroup(): (Option[String], Option[Seq[PartitionAssignmentState]]) = {
-      collectGroupAssignment(opts.options.valueOf(opts.groupOpt))
+    private def dataExistsForGroup(group: String, state: Option[String], numRows: Option[Int]): Boolean = {
```

Comment:
Maybe a better name is `shouldPrintMemberState` or something like that? The third argument still feels a bit odd (in particular when we have to pass `Some(1)` in `printState` below).